code/__HELPERS/radiation.dm

```diff
@@ -0,0 +1,35 @@
+/proc/get_rad_contents(atom/location, list/output=list()) // A special GetAllContents that doesn't search past things with rad insulation
+	. = output
+
+	if(!location)
+		return
+
+	output += location
+
+	var/datum/component/rad_insulation/insulation = location.GetComponent(/datum/component/rad_insulation)
+	if(insulation && insulation.protects)
+		return
+	
+	for(var/i in 1 to location.contents.len)
+		var/static/list/ignored_things = typecacheof(list(/mob/dead, /obj/effect, /obj/docking_port, /turf, /atom/movable/lighting_object))
+		var/atom/thing = location.contents[i]
+		if(ignored_things[thing.type])
+			continue
+		get_rad_contents(thing, output)
```

Comment:
can you make this iterative instead of recursive please

Reply:
The recursive method has been proven faster in #31027